### ISSUE TITLE
[a11y] `SubNav` current indicator not visible on Windows 11 high contrast themes

### ISSUE THREAD

**Issue by stamat** [primer/brand#835]
- Issue: https://github.com/github/accessibility-audits/issues/9835

This happens because high contrast themes enforce backgrounds as certain colors, meaning the background color is impossible to set since it gets overrided on the system level - which in turn means that the underline indicator, instead of being the `currentColor` blends with the background.

It's better to use a border instead of the background color in this case of a current indicator.

To counter this we are temporarily placing an override:

```css
.lp-SubNav li > a > span::after {
  background: none;
  height: 0;
  border-bottom: 2px solid currentColor;
}
```

**Comment by stamat** [comment]
Also it's a good thing that the current page `aria-current="true"` underline is present on mobile also, so the users that don't distinguish color have a visual indicator what's the current page/section.

```css
.lp-SubNav li > a[aria-current="true"] span::after {
  display: block;
  content: "";
}
```

**Comment by rezrah** [comment]
Thanks @stamat 💖 

> It's better to use a border instead of the background color in this case of a current indicator.

During a recent refactor of SubNav `border-color` was preferred over `background-color` for the underline. [Source](https://github.com/primer/brand/blob/main/packages/react/src/SubNav/SubNav.module.css#L153).

Closing this out as the issue should be resolved, with the fix shipping in the next release.
